pupil_src/shared_modules/audio_capture.py

```diff
@@ -196,22 +197,28 @@ def close_recording():
             # Bind nonlocal variables, https://www.python.org/dev/peps/pep-3104/
             nonlocal out_container, out_stream, in_stream, in_frame_size, timestamps, out_frame_num
             if out_container is not None:
-                packet = out_stream.encode(audio_frame)
-                while packet is not None:
-                    out_frame_num += 1
-                    out_container.mux(packet)
-                    packet = out_stream.encode(audio_frame)
+                timestamps.append(timestamp)
+                out_packets = [out_stream.encode(audio_frame)]
+                while out_packets[-1]:
+                    out_packets.append(out_stream.encode(None))
+                for out_packet in out_packets:
+                    if out_packet is not None:
+                        out_container.mux(out_packet)
                 out_container.close()
 
+                out_frame_num = out_stream.frames
                 in_frame_rate = in_stream.rate
                 # in_stream.frame_size does not return the correct value.
                 out_frame_size = out_stream.frame_size
                 out_frame_rate = out_stream.rate
 
-                old_ts_idx = np.arange(0, len(timestamps) * in_frame_size, in_frame_size) / in_frame_rate
                 new_ts_idx = np.arange(0, out_frame_num * out_frame_size, out_frame_size) / out_frame_rate
-                interpolate = interp1d(old_ts_idx, timestamps, bounds_error=False, fill_value='extrapolate')
-                new_ts = interpolate(new_ts_idx)
+                if in_frame_rate != out_frame_rate:
+                    old_ts_idx = np.arange(0, len(timestamps) * in_frame_size, in_frame_size) / in_frame_rate
+                    interpolate = interp1d(old_ts_idx, timestamps, bounds_error=False, fill_value='extrapolate')
+                    new_ts = interpolate(new_ts_idx)
+                else:
+                    new_ts = timestamps[0] + new_ts_idx
```

Comment:
Do I understand it correctly that you are adding up a timestamp and an index here?

Reply:
Adding a fixed offset (stream epoch) to the generated sequence of output timestamps (numpy array)

Reply:
I am still a bit confused about this line. `timestamps` is a Python list that contains floats, if I am not mistaken. And `new_ts_idx` is a numpy array. Is it possible that you meant this line to be `new_ts = np.array(timestamps) + new_ts_idx[0]`?

Reply:
Never mind, I just understood it. 🤦‍♂️

Reply:
No. new_ts_idx contains the exact timestamps calculated with respect to the output frame size, zero based. We just need to add the correct stream epoch which is in fact the value in timestamps[0].